File: backend/JMMinistry.Common/Dtos/DiscipleJourney/DiscipleStepDto.cs
using JMMinistry.Common.Dtos.DiscipleJourney.Enums;

namespace JMMinistry.Common.Dtos.DiscipleJourney
{
    public class DiscipleStepDto
    {
        public string Id { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string? ParentId { get; set; }
        public StepCategory StepCategory { get; set; }
        public bool RequiresCycle { get; set; }
        public bool RequiresAdminApproval { get; set; }
        public IList<string> RequirementIds { get; set; } = [];
        public IList<DiscipleStepDto> SubSteps { get; set; } = [];
    }
}

File: the_vine_hub_app/lib/features/training/domain/disciple_step_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'disciple_step_dto.freezed.dart';
part 'disciple_step_dto.g.dart';

@freezed
abstract class DiscipleStepDto with _$DiscipleStepDto {
  const factory DiscipleStepDto({
    required int id,
    String? name,
    String? description,
    required int stepCategory,
    required bool requiresCycle,
    required bool requiresAdminApproval,
    List<int>? requirementIds,
    int? parentStepId,
    List<DiscipleStepDto>? subSteps,
  }) = _DiscipleStepDto;

  factory DiscipleStepDto.fromJson(Map<String, dynamic> json) =>
      _$DiscipleStepDtoFromJson(json);
}
